demos/keyboard_nav/line_cursor.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The class representing a line cursor.
+ * A line cursor traverses the blocks as if they were
+ * lines of code in a text editor. 
+ * Previous and next go up and down lines. In and out go
+ * through the elements in a line.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+'use strict';
+
+
+/**
+ * Class for a line cursor.
+ * This will allow the user to get to all nodes in the AST by hitting next or
+ * previous.
+ * @constructor
+ * @extends {Blockly.LineCursor}
```

Comment:
Extends itself? I think you mean extends BasicCursor

Reply:
Woops, fixed!